opslevel/resource_opslevel_service_tag.go

```diff
@@ -238,5 +239,45 @@ func (serviceTagResource *ServiceTagResource) Delete(ctx context.Context, req re
 }
 
 func (serviceTagResource *ServiceTagResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
-	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+	if !isTagValid(req.ID) {
+		resp.Diagnostics.AddError(
+			"Invalid format for given Import Id",
+			fmt.Sprintf("Id expected to be formatted as '<service-id>:<tag-id>'. Given '%s'", req.ID),
+		)
+	}
+
+	ids := strings.Split(req.ID, ":")
+	serviceId := ids[0]
+	tagId := ids[1]
+
+	service, err := serviceTagResource.client.GetTaggableResource(opslevel.TaggableResourceService, serviceId)
+	if err != nil || service == nil {
```

Comment:
This is another major opslevel-go painpoint that we need to solve, there's way too much cognitive overhead from fetching certain resources and understanding whether or not you can rely on the returned `error` to work appropriately for checking if an API call failed.

Reply:
A refactor toward something like this would be nice but there are so many pressing tasks that I don't know when we would get to it